azure-quantum/azure/quantum/_authentication/_default.py

```diff
@@ -97,7 +99,18 @@ def _initialize_credentials(self):
             credentials.append(VisualStudioCodeCredential(authority=self.authority, tenant_id=self.tenant_id))
             credentials.append(AzureCliCredential(tenant_id=self.tenant_id))
             credentials.append(AzurePowerShellCredential(tenant_id=self.tenant_id))
-            credentials.append(InteractiveBrowserCredential(authority=self.authority, tenant_id=self.tenant_id))
+            # Before trying other credential types, try to use already cached token.
+            credentials.append(SharedTokenCacheCredential(authority=self.authority))
```

Comment:
On Linux, when passing `self.tenant_id` to the `SharedTokenCacheCredential`, it fails due to a mismatch of this tenant_id with tenant_id from the cache.  Still haven't figured it out.

Reply:
even if the `tenant_id` is not passed, we automatically fetch it anyway by calling [`self._discover_tenant_id_()`](https://github.com/microsoft/azure-quantum-python/blob/01c2dd05a34e5d6368e6690a279d0ccbf5a51026/azure-quantum/azure/quantum/_authentication/_default.py#L88)

Reply:
But that's for the constructor of the `_DefaultAzureCredential`, right? I meant passing the `tenant_id` to the `SharedTokenCacheCredential` itself.

Reply:
Also, since the `InteractiveBrowserCredential` already has support for caching, I'm confused why we need to add the `SharedTokenCacheCredential` in the list of credentials.

Reply:
Ok, I think I get it.
If we enable the caching in the `InteractiveBrowserCredential` it will fail even if the browser authentication would succeed but accessing the cache fails.

Reply:
Please see my latest commit.
Using our own cache will mitigate issues with tokens cached by other apps.
As Kirill mentioned, if there was a change of `authority` or `tenant_id`, the `SharedTokenCacheCredential` will be skipped and the user can authenticate with the `InteractiveBrowserCredential` (which will store a new token for these new parameters).